jwk/jwks.go

```diff
@@ -0,0 +1,22 @@
+package jwk
+
+import "fmt"
+
+type JWKSet struct {
+	Keys   []JWK `json:"keys"`
+	keyMap map[string]*JWK
+}
+
+func (j *JWKSet) Get(kid string) (*JWK, error) {
+	if j.keyMap == nil {
+		j.keyMap = make(map[string]*JWK)
+		for _, key := range j.Keys {
+			j.keyMap[key.KeyID] = &key
+		}
+	}
+	key, ok := j.keyMap[kid]
```

Comment:
If `!ok` then `key == nil`.

Just `key := j.keyMap[kid]` and `if key == nil { no key found }`. After all, we don’t want to ever return `nil, nil` but we could with this implementation.